pkg/labels/consul_aggregator.go

```diff
@@ -184,7 +154,56 @@ func (c *consulAggregator) Aggregate() {
 		case <-c.aggregatorQuit:
 			return
 		case <-loopTime:
+			// we purposely don't case outErrors here, since loopTime lets us
+			// back off of Consul watches. If an error repeatedly were occurring,
+			// we could end up in a nasty busy loop.
+		}
+	}
+}
+
+func (c *consulAggregator) fillCache(pairs api.KVPairs) {
+	cache := make([]Labeled, len(pairs))
+	c.labeledCache = cache
+	for i, kvp := range pairs {
+		val, err := convertKVPToLabeled(kvp)
+		if err != nil {
+			c.logger.WithErrorAndFields(err, logrus.Fields{
+				"key":   kvp.Key,
+				"value": string(kvp.Value),
+			}).Errorln("Invalid key encountered, skipping this value")
+			continue
 		}
+		c.labeledCache[i] = val
+	}
+}
 
+// this must be called within the watcherLock mutex.
+func (c *consulAggregator) doMatch(watcher *selectorWatch) {
```

Comment:
nit: i'd maybe call this `sendMatches` or something because it does the actual sending on the channel, wasn't obvious from the name.